### PR TITLE
update shiki to v0.12.1

### PR DESCRIPTION
**Pull request by B2o5T**
fixes https://github.com/shuding/nextra/issues/934

### CODE REVIEW

File: .changeset/plenty-experts-push.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'nextra': patch
+---
+
+update rehype-pretty-code to v0.4.0
```
Comment (B2o5T) [inline .changeset/plenty-experts-push.md:5]:
```suggestion
update rehype-pretty-code to v0.5.1
```

**Comment by shuding** [comment]
Is there a performance regression in these two libraries? We are getting this error from the deployment...

> Your deployment's build step did not complete within the maximum of 45 minutes

**Comment by B2o5T** [comment]
> Is there a performance regression in these two libraries? We are getting this error from the deployment...
> 
> > Your deployment's build step did not complete within the maximum of 45 minutes

@shuding there are some memory leaks issues in newly version of rehype code, I already reported it, will convert this PR as a draft until they fix it

File: .changeset/plenty-experts-push.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'nextra': patch
+---
+
+update rehype-pretty-code to v0.5.1, shiki to v0.11.1
```
Comment (B2o5T) [inline .changeset/plenty-experts-push.md:5]:
```suggestion
update rehype-pretty-code to v0.6.0-beta.0, shiki to v0.11.1
```

**Comment by B2o5T** [comment]
@shuding it's ready to merge but shell code blocks looks differently
now
image
before
image

**Comment by shuding** [comment]
I think that's fine as we use `bash`

**Pull request merged**